News-Android-App/src/main/java/de/luhmer/owncloudnewsreader/SettingsActivity.java

```diff
@@ -324,9 +324,16 @@ public boolean onPreferenceChange(Preference preference, Object value) {
 				int index = listPreference.findIndexOfValue(stringValue);
 
 				// Set the summary to reflect the new value.
-				preference
-						.setSummary(index >= 0 ? listPreference.getEntries()[index]
-								: null);
+				preference.setSummary(index >= 0 ? listPreference.getEntries()[index] : null);
+
+				// only enable black-bg setting if light or auto theme is selected
+				if(preference.getKey().equals("sp_app_theme")) {
```

Comment:
please use constant here: SP_APP_THEME
```suggestion
				if(preference.getKey().equals(SP_APP_THEME)) {
```

Reply:
I think `getKey()` will never be null - however just a small tip that I learned a couple weeks ago. Constants should be on the left side of comparison statement. Why? 
In case `preference.getKey()` returns `null`, calling `.equals(..)` will fail with a Null-Pointer-Exception. If you write it like this: `SP_APP_THEME.equals(preference.getKey())` it won't fail, it'll just say no, it's not equal.

We don't have to change anything though, just a thought.

Reply:
Very valid point - adopted, plus also fixed the other instance where I didn't use the declared constant